src/GitVersionCore.Tests/BuildServers/VsoAgentTests.cs

```diff
@@ -1,16 +1,35 @@
-﻿using GitVersion;
+﻿using System;
+using GitVersion;
+using GitVersionCore.Tests;
 using NUnit.Framework;
 using Shouldly;
 
 [TestFixture]
 public class VsoAgentTests
 {
+
+    string key = "BUILD_BUILDNUMBER";
+
+    private void SetEnvironmentVariableForTest()
+    {
+        Environment.SetEnvironmentVariable(key, "Some Build_Value $(GitVersion_FullSemVer)", EnvironmentVariableTarget.Process);
```

Comment:
I was trying to figure out what'd make sense as test input for that. Those variables that TFS does recognize will be pre-expanded though.